src/librustc_save_analysis/dump_visitor.rs

```diff
@@ -139,6 +145,138 @@ impl<'l, 'tcx: 'l, 'll, D: Dump + 'll> DumpVisitor<'l, 'tcx, 'll, D> {
         self.dumper.crate_prelude(data);
     }
 
+    pub fn dump_borrow_analysis(&mut self) {
+        let borrow_analysis = self.save_ctxt.borrow_analysis_map.drain().collect::<Vec<_>>();
+        for (def_id, analysis_result) in borrow_analysis {
+            self.process_borrow_analysis(def_id, analysis_result);
+        }
+    }
+
+    fn process_borrow_analysis(&mut self, def_id: DefId, mut analysis_result: AnalysisResult<'l, 'tcx>) {
+        let mut safe_loans = {
+            let mut safe_loans = vec![];
+            mem::swap(&mut safe_loans, &mut analysis_result.safe_loans);
```

Comment:
Here I'm doing the `mem::swap` and `into_iter` calls to avoid cloning all of the data (all scopes, borrows and moves for the entire crate). Is that necessary? I'm not sure if I'm just being overly paranoid about the memory usage getting too high.